### PR TITLE
fix: 修复画布宽度设置问题

### LINKED ISSUE #120: 🐛 [Bug]: 画布设置在pc端宽度设置为1440px时，会自动切换为大屏下且宽度被设置成了1920px
### Environment

chrome

### Version

latest

### Version

latest

### Link to minimal reproduction

~HS6ZMD)1)G_GV F$%M $XM

### Step to reproduce

顶部工具栏画布设置选择pc端宽度设置为1440px后，回车确认会自动切换为大屏下且宽度被设置成了1920px

### What is expected

希望画布实际宽度及输入框显示的是我输入的值。

### What is actually happening

No response

### Any additional comments (optional)

No response

### PR DESCRIPTION
**Pull request by gene9831**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: #120 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/toolbars/media/src/Main.vue [hunk 1]
```diff
@@ -314,29 +312,28 @@ export default {
...
         state.width = 1920
-        prevWidthVal = 1920
       } else {
-        prevWidthVal = val
+        state.width = val
       }
+      prevWidthVal = state.width
 
       switch (true) {
         case 1200 < Number(state.width) && Number(state.width) <= 1920:
```
Comment (chilingling) [inline packages/toolbars/media/src/Main.vue]: `Number(state.width)` 这个可以存一个变量，避免每个 case 里面都去转化一次

File: packages/toolbars/media/src/Main.vue [hunk 2]
```diff
@@ -307,36 +302,34 @@ export default {
       })
     }
     const widthChange = (val) => {
```
Comment (chilingling) [inline packages/toolbars/media/src/Main.vue:314]: 一整个函数圈复杂度高了，建议抽取逻辑出来

**Review by chilingling — CHANGES_REQUESTED** [review]

**Review by chilingling — DISMISSED** [review]

File: packages/toolbars/media/src/Main.vue [hunk 3]
```diff
@@ -298,54 +295,43 @@ export default {
...
+      { type: 'lanMobile', min: 480, max: 768 },
+      { type: 'tablet', min: 768, max: 992 },
+      { type: 'desktop', min: 992, max: 1200 },
+      { type: 'mdx', min: 1200, max: 1920 }
+    ]
+
     const widthChange = (val) => {
-      const reg = '^[0-9]*$'
-      const dimension = useLayout().getDimension()
+      const reg = '^[0-9]+$'
```
Comment (wenmine) [inline packages/toolbars/media/src/Main.vue]: 这个正则可以抽成一个全局constant吗，下面也用到了相同的正则，可以抽成一个复用。

Comment (gene9831) [reply packages/toolbars/media/src/Main.vue]: 宽度值和缩放值显示出来的格式不一致。缩放值可以带小数，应该修改缩放的正则表达式：`'^[0-9]+(\.[0-9]+)?$'`
image

File: packages/toolbars/media/src/Main.vue [hunk 4]
```diff
@@ -298,54 +295,43 @@ export default {
...
-      }
+      const width = Number(state.width)
+      const type = breakpoints.find((item) => item.min <= width && width <= item.max)?.type || 'desktop'
+      activeView(width, type)
     }
 
     const scaleChange = (val) => {
       const item = mediaMap['mdx']
-      const reg = '^[0-9]*$'
+      const reg = '^[0-9]+(\\.[0-9]+)?$'
```
Comment (hexqi) [inline packages/toolbars/media/src/Main.vue]: 建议不用字符串，直接写正则表达式： `/^\d+(\.\d+)?$/`

File: packages/toolbars/media/src/Main.vue [hunk 5]
```diff
@@ -370,22 +353,24 @@ export default {
       setCanvasType(value ? 'absolute' : 'normal')
     }
 
-    watchEffect(
-      () => {
-        const mode = dimension.value.deviceType || 'desktop'
+    watchEffect(() => {
+      const deviceType = dimension.value.deviceType
```
Comment (hexqi) [inline packages/toolbars/media/src/Main.vue]: 这里看起来没必要用watchEffect，watch dimension.value.deviceType就行了

File: packages/toolbars/media/src/Main.vue [hunk 6]
```diff
@@ -370,22 +353,24 @@ export default {
...
-      },
-      { flush: 'post' }
-    )
+    watchEffect(() => {
+      state.scaleValue = scale.value.toFixed(2)
+      prevScaleVal = scale.value
+    })
 
     watch(
       () => useLayout().getDimension().width,
```
Comment (hexqi) [inline packages/toolbars/media/src/Main.vue]: 可以跟其他地方统一，写成：dimension.value.width

File: packages/toolbars/media/src/Main.vue [hunk 7]
```diff
@@ -298,54 +295,43 @@ export default {
...
-      state.activeIndex = mediaMap[type].index
       useLayout().setDimension({
         deviceType: item.view,
         width: val ? `${val}px` : item.width,
         minWidth: item.minWidth,
         maxWidth: item.maxWidth
       })
     }
+
+    const breakpoints = [
```
Comment (hexqi) [inline packages/toolbars/media/src/Main.vue:306]: image
可以记一个历史遗留问题

**Review by hexqi — APPROVED** [review]

File: packages/toolbars/media/src/Main.vue (hunk 7 above)
Comment (gene9831) [reply packages/toolbars/media/src/Main.vue:306]: #172 跟踪此问题

**Review by chilingling — APPROVED** [review]

**Pull request merged**